fix: Fix the problem of null judgment error

huangjunjia opened the pull request:
## Description

In the current null and nnull judgments, undefined will be erroneously judged to be true.

Adjusted the judgment logic, the judgment rules of null and nnull return true or false according to whether the value is nil.

And complements the logic of recursive judgment.

self-transition commented:
CI failed

huangjunjia commented:
@self-transition fixed

self-transition commented:
IsNil adds an extra check for undefined, so do you need to add a unit test for it?

I'm also wondering if the previous `_.get` call returns `undefined` if a non-existent key is passed in. and if that's what's expected in this case.

huangjunjia commented:
> IsNil adds an extra check for `undefined`, so do you need to add a unit test for it?
> 
> I'm also wondering if the previous `_.get` call returns `undefined` if a non-existent key is passed in. and if that's what's expected in this case.

I have adjusted the judgment logic based on your response. 

Checks if path is a direct property of object before judge conditions, and added the relevant unit tests.

PTAL

self-transition commented on src/utils/filter-data.ts:
```diff
@@ -92,10 +86,18 @@ export function evaluateFilter(
-      return fieldValue === null;
-    case 'nnull':
-      return fieldValue !== null;
+    case 'null': {
```
Is it possible to put this judgment at the beginning of the function, so that when path is a direct property of object, it doesn't do anything and returns the original value?

huangjunjia replied:
you're right! I think it should return false when path isn't direct property of object, because it mean that data is not valid.

self-transition reviewed (approved):
LGTM
